src/ring_theory/multiplicity.lean

```diff
@@ -169,6 +169,20 @@ lemma eq_of_associated_right {a b c : α} (h : associated b c) :
 le_antisymm (multiplicity_le_multiplicity_of_dvd_right (dvd_of_associated h))
   (multiplicity_le_multiplicity_of_dvd_right (dvd_of_associated h.symm))
 
+lemma multiplicity.unit_left {a: α}
+  (u: units α): multiplicity (u: α) a = ⊤ :=
+begin
+  rw ← multiplicity.eq_of_associated_left unit_associated_one,
+  exact multiplicity.one_left a,
+end
```

Comment:
We already have:

https://github.com/leanprover-community/mathlib/blob/134531919073f222674d273a91389a559aad0f88/src/ring_theory/multiplicity.lean#L130-L131

Can you move this lemma to just below it, and prove it using `is_unit_unit` and `multiplicity.multiplicity_unit`?

Reply:
Actually I hesitate a bit about this.
Of course, it would make for a simpler, more elementary proof.
But my problem is that this technique does not apply to `multiplicity.unit_right`, and performing this change would break the symmetry that exists between the two proofs, and I guess it would make more spaghetti in the file.
So what should I do ?

Reply:
You could partially restore the symmetry by adding a lemma:
```
lemma multiplicity.is_unit_right {a b : α} (ha: ¬is_unit a) (hb : is_unit b) : multiplicity a b = 0 :=
```
and using that to prove `multiplicity.unit_right`.

Reply:
Well, this shortened the proofs ! :tada: